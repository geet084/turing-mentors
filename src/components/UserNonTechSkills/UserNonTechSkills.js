import React, { Component } from 'react';
import { CreateCheckbox } from '../CreateCheckbox/CreateCheckbox';

export class UserNonTechSkills extends Component {
  state = {
    nonTechSkills: {
      StressManagement: false,
      PublicSpeaking: false,
      Resumes: false,
      Technical_Interview: false,
      Parenting: false,
      Wellness: false,
    }
  }

  submitForm = (e) => {
    e.preventDefault();
    this.props.updateUserInfo([this.state, 'complete']);
  }

  goBack = (e) => {
    e.preventDefault();
    this.props.updateUserInfo([this.state, 'userTechSkills']);
  }

  checkBoxes = ({ target }) => {
    let nonTechSkills = { ...this.state.nonTechSkills };
    nonTechSkills[target.name] = !nonTechSkills[target.name]

    this.setState({ nonTechSkills })
  }

  render() {
    const mentor = this.props.user === 'Mentor';
    const { nonTechSkills } = this.state;
    const checkBoxes = Object.keys(nonTechSkills).map((skill, index) => {
      return <CreateCheckbox key={skill} field={skill} name={skill} value={index} checkBoxes={this.checkBoxes} />
    })

    return (
      <div className="form-page">
        <p className="pages">Mentor NON Tech skills</p>
        <form>
          {checkBoxes}
        </form>
        <button className="next-btn" onClick={this.goBack}>Back</button>
        <button className="next-btn" onClick={this.submitForm}>Submit</button>
        <span className="pages">6 of {mentor ? '6' : '4'}</span>
      </div>
    )
  }
}

export default UserNonTechSkills;