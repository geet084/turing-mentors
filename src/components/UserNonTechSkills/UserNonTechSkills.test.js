import React from 'react';
import { shallow } from 'enzyme';
import { UserNonTechSkills } from './UserNonTechSkills';

describe('UserNonTechSkills', () => {
  let wrapper;
  const mockProps = {
    updateUserInfo: jest.fn(),
    user: 'Mentee',
  };
  const mockState = {
    non_tech_skills: []
  };

  beforeEach(() => {
    wrapper = shallow(<UserNonTechSkills {...mockProps} />);
  });

  describe('initial snapshot', () => {
    it('should match the correct default snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the correct populated non_tech_skills snapshot', () => {
      wrapper.setState({ non_tech_skills: ['1'] });
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the correct user snapshot', () => {
      wrapper.setProps({ user: 'Mentor' });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('submitForm', () => {
    it('should handle the form submission', () => {
      const mockEvent = { preventDefault: () => { } };
      const expected = [mockState, 'complete'];

      wrapper.instance().submitForm(mockEvent);

      expect(mockProps.updateUserInfo).toHaveBeenCalledWith(expected);
    });
  });

  describe('goBack', () => {
    it('should handle going back a page', () => {
      const mockEvent = { preventDefault: () => { } };
      const expected = [mockState, 'userTechSkills'];

      wrapper.instance().goBack(mockEvent);

      expect(mockProps.updateUserInfo).toHaveBeenCalledWith(expected);
    });
  });

  describe('checkBoxes', () => {
    it('should update the selected input in state', () => {
      const mockEvent = { target: { value: 3 } };
      const initial = { non_tech_skills: [] };
      const expected = { non_tech_skills: [3] };

      expect(wrapper.state()).toEqual(initial);

      wrapper.instance().checkBoxes(mockEvent);
      expect(wrapper.state()).toEqual(expected);
    });

    it('should update the deselected input in state', () => {
      const mockEvent = { target: { value: 3 } };
      const changed = { non_tech_skills: [3] };
      const expected = { non_tech_skills: [] };

      expect(wrapper.state()).toEqual(expected);

      wrapper.instance().checkBoxes(mockEvent);
      expect(wrapper.state()).toEqual(changed);

      wrapper.instance().checkBoxes(mockEvent);
      expect(wrapper.state()).toEqual(expected);

    });
  });
});