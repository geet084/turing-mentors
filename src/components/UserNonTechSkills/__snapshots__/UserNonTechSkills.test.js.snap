// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNonTechSkills should match the correct snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserNonTechSkills
    updateUserInfo={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="pages"
        >
          Mentor NON Tech skills
        </p>,
        <form>
          <CreateCheckbox
            checkBoxes={[Function]}
            field="stress_management"
            name="stress_management"
            value={1}
          />
          <CreateCheckbox
            checkBoxes={[Function]}
            field="public_speaking"
            name="public_speaking"
            value={2}
          />
          <CreateCheckbox
            checkBoxes={[Function]}
            field="resumes"
            name="resumes"
            value={3}
          />
          <CreateCheckbox
            checkBoxes={[Function]}
            field="technical_interviews"
            name="technical_interviews"
            value={4}
          />
          <CreateCheckbox
            checkBoxes={[Function]}
            field="parenting"
            name="parenting"
            value={5}
          />
          <CreateCheckbox
            checkBoxes={[Function]}
            field="wellness"
            name="wellness"
            value={6}
          />
        </form>,
        <button
          className="next-btn"
          onClick={[Function]}
        >
          Back
        </button>,
        <button
          className="next-btn"
          onClick={[Function]}
        >
          Submit
        </button>,
        <span
          className="pages"
        >
          6 of 
          4
        </span>,
      ],
      "className": "form-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Mentor NON Tech skills",
          "className": "pages",
        },
        "ref": null,
        "rendered": "Mentor NON Tech skills",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <CreateCheckbox
              checkBoxes={[Function]}
              field="stress_management"
              name="stress_management"
              value={1}
            />,
            <CreateCheckbox
              checkBoxes={[Function]}
              field="public_speaking"
              name="public_speaking"
              value={2}
            />,
            <CreateCheckbox
              checkBoxes={[Function]}
              field="resumes"
              name="resumes"
              value={3}
            />,
            <CreateCheckbox
              checkBoxes={[Function]}
              field="technical_interviews"
              name="technical_interviews"
              value={4}
            />,
            <CreateCheckbox
              checkBoxes={[Function]}
              field="parenting"
              name="parenting"
              value={5}
            />,
            <CreateCheckbox
              checkBoxes={[Function]}
              field="wellness"
              name="wellness"
              value={6}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "stress_management",
            "nodeType": "class",
            "props": Object {
              "checkBoxes": [Function],
              "field": "stress_management",
              "name": "stress_management",
              "value": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "public_speaking",
            "nodeType": "class",
            "props": Object {
              "checkBoxes": [Function],
              "field": "public_speaking",
              "name": "public_speaking",
              "value": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "resumes",
            "nodeType": "class",
            "props": Object {
              "checkBoxes": [Function],
              "field": "resumes",
              "name": "resumes",
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "technical_interviews",
            "nodeType": "class",
            "props": Object {
              "checkBoxes": [Function],
              "field": "technical_interviews",
              "name": "technical_interviews",
              "value": 4,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "parenting",
            "nodeType": "class",
            "props": Object {
              "checkBoxes": [Function],
              "field": "parenting",
              "name": "parenting",
              "value": 5,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "wellness",
            "nodeType": "class",
            "props": Object {
              "checkBoxes": [Function],
              "field": "wellness",
              "name": "wellness",
              "value": 6,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Back",
          "className": "next-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Back",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "className": "next-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "6 of ",
            "4",
          ],
          "className": "pages",
        },
        "ref": null,
        "rendered": Array [
          "6 of ",
          "4",
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="pages"
          >
            Mentor NON Tech skills
          </p>,
          <form>
            <CreateCheckbox
              checkBoxes={[Function]}
              field="stress_management"
              name="stress_management"
              value={1}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="public_speaking"
              name="public_speaking"
              value={2}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="resumes"
              name="resumes"
              value={3}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="technical_interviews"
              name="technical_interviews"
              value={4}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="parenting"
              name="parenting"
              value={5}
            />
            <CreateCheckbox
              checkBoxes={[Function]}
              field="wellness"
              name="wellness"
              value={6}
            />
          </form>,
          <button
            className="next-btn"
            onClick={[Function]}
          >
            Back
          </button>,
          <button
            className="next-btn"
            onClick={[Function]}
          >
            Submit
          </button>,
          <span
            className="pages"
          >
            6 of 
            4
          </span>,
        ],
        "className": "form-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Mentor NON Tech skills",
            "className": "pages",
          },
          "ref": null,
          "rendered": "Mentor NON Tech skills",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <CreateCheckbox
                checkBoxes={[Function]}
                field="stress_management"
                name="stress_management"
                value={1}
              />,
              <CreateCheckbox
                checkBoxes={[Function]}
                field="public_speaking"
                name="public_speaking"
                value={2}
              />,
              <CreateCheckbox
                checkBoxes={[Function]}
                field="resumes"
                name="resumes"
                value={3}
              />,
              <CreateCheckbox
                checkBoxes={[Function]}
                field="technical_interviews"
                name="technical_interviews"
                value={4}
              />,
              <CreateCheckbox
                checkBoxes={[Function]}
                field="parenting"
                name="parenting"
                value={5}
              />,
              <CreateCheckbox
                checkBoxes={[Function]}
                field="wellness"
                name="wellness"
                value={6}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "stress_management",
              "nodeType": "class",
              "props": Object {
                "checkBoxes": [Function],
                "field": "stress_management",
                "name": "stress_management",
                "value": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "public_speaking",
              "nodeType": "class",
              "props": Object {
                "checkBoxes": [Function],
                "field": "public_speaking",
                "name": "public_speaking",
                "value": 2,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "resumes",
              "nodeType": "class",
              "props": Object {
                "checkBoxes": [Function],
                "field": "resumes",
                "name": "resumes",
                "value": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "technical_interviews",
              "nodeType": "class",
              "props": Object {
                "checkBoxes": [Function],
                "field": "technical_interviews",
                "name": "technical_interviews",
                "value": 4,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "parenting",
              "nodeType": "class",
              "props": Object {
                "checkBoxes": [Function],
                "field": "parenting",
                "name": "parenting",
                "value": 5,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "wellness",
              "nodeType": "class",
              "props": Object {
                "checkBoxes": [Function],
                "field": "wellness",
                "name": "wellness",
                "value": 6,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Back",
            "className": "next-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Back",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "next-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "6 of ",
              "4",
            ],
            "className": "pages",
          },
          "ref": null,
          "rendered": Array [
            "6 of ",
            "4",
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
